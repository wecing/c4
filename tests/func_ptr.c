extern int printf(const char *s, ...);

void f() {
  printf("f\n");
}

static void f2(void);

typedef void (*FP)(void);

int main() {
  FP fp = f;
  fp();

  fp = f;
  fp();
  fp = &f;
  fp();

  (*fp)();

  fp = f2;
  fp();

  return 0;
}

static void f2() {
  printf("f2\n");
}
