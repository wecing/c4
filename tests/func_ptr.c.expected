f
f
f
f
f2
